File: urbanairship-core/src/test/java/com/urbanairship/javascript/JavaScriptEnvironmentTest.java
package com.urbanairship.javascript;

import com.urbanairship.BaseTestCase;
import com.urbanairship.json.JsonMap;
import com.urbanairship.json.JsonValue;

import org.junit.Test;

import java.util.Arrays;

import androidx.test.core.app.ApplicationProvider;

import static org.junit.Assert.assertTrue;

public class JavaScriptEnvironmentTest extends BaseTestCase {

    @Test
    public void testStringGetter() {
        JavaScriptEnvironment environment = JavaScriptEnvironment.newBuilder()
                             .addGetter("cool", "neat")
                             .build();

        String javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext());
        String expected = "_UAirship.cool = function(){return \"neat\";};";
        assertTrue(javaScript.contains(expected));
    }

    @Test
    public void testNumberGetter() {
        JavaScriptEnvironment environment = JavaScriptEnvironment.newBuilder()
                                                                 .addGetter("rad", 100)
                                                                 .build();

        String javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext());
        String expected = "_UAirship.rad = function(){return 100;};";
        assertTrue(javaScript.contains(expected));
    }

    @Test
    public void testJsonMapGetter() {
        JsonMap map = JsonMap.newBuilder()
                .put("cool", "story")
                .build();

        JavaScriptEnvironment environment = JavaScriptEnvironment.newBuilder()
                                                                 .addGetter("jsonMap", map)
                                                                 .build();

        String javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext());
        String expected = "_UAirship.jsonMap = function(){return {\"cool\":\"story\"};};";
        assertTrue(javaScript.contains(expected));
    }

    @Test
    public void testJsonArrayGetter() {
        JsonValue value = JsonValue.wrapOpt(Arrays.asList("foo", "bar"));

        JavaScriptEnvironment environment = JavaScriptEnvironment.newBuilder()
                                                                 .addGetter("jsonArray", value)
                                                                 .build();

        String javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext());
        String expected = "_UAirship.jsonArray = function(){return [\"foo\",\"bar\"];};";
        assertTrue(javaScript.contains(expected));
    }

}

File: urbanairship-core/src/test/java/com/urbanairship/javascript/JavaScriptEnvironmentTest.kt
package com.urbanairship.javascript

import androidx.test.core.app.ApplicationProvider
import androidx.test.ext.junit.runners.AndroidJUnit4
import com.urbanairship.json.JsonValue
import com.urbanairship.json.jsonMapOf
import org.junit.Assert
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
public class JavaScriptEnvironmentTest {

    @Test
    public fun testStringGetter() {
        val environment = JavaScriptEnvironment.newBuilder()
            .addGetter("cool", "neat")
            .build()

        val javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext())
        val expected = "_Airship.cool = function(){return \"neat\";};"
        Assert.assertTrue(javaScript.contains(expected))
    }

    @Test
    public fun testNumberGetter() {
        val environment = JavaScriptEnvironment.newBuilder()
            .addGetter("rad", 100)
            .build()

        val javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext())
        val expected = "_Airship.rad = function(){return 100;};"
        Assert.assertTrue(javaScript.contains(expected))
    }

    @Test
    public fun testJsonMapGetter() {
        val map = jsonMapOf("cool" to "story")

        val environment = JavaScriptEnvironment.newBuilder()
            .addGetter("jsonMap", map)
            .build()

        val javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext())
        val expected = "_Airship.jsonMap = function(){return {\"cool\":\"story\"};};"
        Assert.assertTrue(javaScript.contains(expected))
    }

    @Test
    public fun testJsonArrayGetter() {
        val value = JsonValue.wrapOpt(listOf("foo", "bar"))

        val environment = JavaScriptEnvironment.newBuilder()
            .addGetter("jsonArray", value)
            .build()

        val javaScript = environment.getJavaScript(ApplicationProvider.getApplicationContext())
        val expected = "_Airship.jsonArray = function(){return [\"foo\",\"bar\"];};"
        Assert.assertTrue(javaScript.contains(expected))
    }
}
